router/http/src/main/scala/io/buoyant/router/http/HeaderIdentifier.scala

```diff
@@ -0,0 +1,26 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.http.Request
+import com.twitter.finagle.{Path, Dtab}
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.util.{Try, Future}
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.{UnidentifiedRequest, IdentifiedRequest, RequestIdentification}
+
+case class HeaderIdentifier(
+  prefix: Path,
+  header: String,
+  baseDtab: () => Dtab = () => Dtab.base
+) extends RoutingFactory.Identifier[Request] {
+
+  def apply(req: Request): Future[RequestIdentification[Request]] = {
+    req.headerMap.get(header) match {
+      case Some(value) =>
+        val path = Try(Path.read(value)).getOrElse(Path.Utf8(value))
```

Comment:
Who would ever come up with such an awful idea?

We could support this IFF the value contains only showable path chars...  Not sure it's worth it? Up to you.